server/routes/unusedDeductionRoutes.ts

```diff
@@ -110,12 +116,33 @@ export default class UnusedDeductionRoutes {
     const { nomsId, addOrEdit } = req.params
     const { prisonerNumber } = res.locals.prisoner
     const { username } = res.locals.user
-    const sessionAdjustment = this.adjustmentsStoreService.getOnly(req, nomsId)
+    const { bookingId } = res.locals.prisoner
+    const startOfSentenceEnvelope = await this.prisonerService.getStartOfSentenceEnvelope(bookingId, username)
+    const adjustments = await this.adjustmentsService.findByPerson(
+      nomsId,
+      startOfSentenceEnvelope.earliestSentence,
+      username,
+    )
+
+    const unusedDeductionAdjustment = adjustments
```

Comment:
Can this all be tidied up abit. Its not immediately clear, that not having a session adjustment means that your doing a deletion.

```
const sessionAdjustment =
      this.adjustmentsStoreService.getOnly(req, nomsId) 
const isDeletion = !sessionAdjustment

await this.adjustmentsService.setUnusedDaysManually(prisonerNumber, isDeletion ? 0 : sessionAdjustment.days, username)

let messageDays: number
let messageAction: string

if (isDeletion) {
const unusedDeductionAdjustment = adjustments.filter(it => it.source !== 'NOMIS')
      .find(it => it.adjustmentType === 'UNUSED_DEDUCTIONS')
      messageAction = 'REMOVE'
      messageDays = unusedDeductionAdjustment.effectiveDays
} else {
messageAction = addOrEdit === 'edit' ? 'UPDATE' : 'CREATE'
messageDays = sessionAdjustment.days
}

```

Reply:
Also if you edit the unused deductions, enter a number, on confirm and save click back then click delete. You'll have a session adjustment, but the users clicked back. So you might need to be more explicit that the users trying to delete.